tests/compilation_cache_test.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+import hashlib
+from jax.experimental.compilation_cache import compilation_cache as cc
+import jax
+import jax.test_util as jtu
+import numpy as np
+import random
+import unittest
+
+class CompilationCacheTest(jtu.JaxTestCase):
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_compile_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.filled_compile_options()
+    filled_hash1 = self.get_hashed_value(cc._hash_compile_options, compile_options_filled)
+    filled_hash2 = self.get_hashed_value(cc._hash_compile_options, compile_options_filled)
+    not_filled_hash3 = self.get_hashed_value(cc._hash_compile_options, compile_options_not_filled)
+    self.assertEqual(filled_hash1, filled_hash2)
+    self.assertNotEqual(filled_hash1, not_filled_hash3)
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_executable_build_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.filled_compile_options()
+    filled_hash1 = self.get_hashed_value(cc._hash_executable_build_options,
+                   compile_options_filled.executable_build_options)
+    filled_hash2 = self.get_hashed_value(cc._hash_executable_build_options,
+                   compile_options_filled.executable_build_options)
+    not_filled_hash3 = self.get_hashed_value(cc._hash_executable_build_options,
+                       compile_options_not_filled.executable_build_options)
+    self.assertEqual(filled_hash1, filled_hash2)
+    self.assertNotEqual(filled_hash1, not_filled_hash3)
+
+  def test_debug_options(self):
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    filled_hash1 = self.get_hashed_value(cc._hash_debug_options,
```

Comment:
`filled_hash` doesn't really make sense here, since there's no concept of filled vs. not filled for debug options. Please rename the variables in this test + below tests that also don't have a "filled" concept (I think all of the below ones). You can just use `hash1`, `hash2`, etc., or something more descriptive if you feel inspired.